src/Common/tests/System/DirectoryServices/LdapConfiguration.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Xml.Linq;
+
+namespace System.DirectoryServices.Tests
+{
+    internal class LdapConfiguration
+    {
+        private LdapConfiguration(string serverName, string domain, string userName, string password, string port, AuthenticationTypes at)
+        {
+            ServerName = serverName;
+            Domain = domain;
+            UserName = userName;
+            Password = password;
+            Port = port;
+            AuthenticationTypes = at;
+        }
+
+        private static LdapConfiguration s_ldapConfiguration = GetConfiguration("LDAP.Configuration.xml");
+
+        internal static LdapConfiguration Configuration =>  s_ldapConfiguration;
+
+        internal string ServerName { get; set; }
+        internal string UserName { get; set; }
+        internal string Password { get; set; }
+        internal string Port { get; set; }
+        internal string Domain { get; set; }
+        internal AuthenticationTypes AuthenticationTypes { get; set; }
+        internal string LdapPath => String.IsNullOrEmpty(Port) ? $"LDAP://{ServerName}/{Domain}" : $"LDAP://{ServerName}:{Port}/{Domain}";
```

Comment:
This format is very specific to the System.DirectoryServices* assembly but doesn't apply to System.DirectoryServices.Protocols  We may want to reuse this config class for all the S.DS.* assemblies to maybe this assembly specific logic should be moved to the assembly specific tests.

Reply:
When adding tests to System.DirectoryServices.Protocols, we can look at what configuration is used there and then can update the configuration class to support it. or to use another specific configuration class.